### ISSUE TITLE
Potentially incorrect/useless test case for LiveVariablesAnalysis

### ISSUE THREAD

**Issue by CamWass** [google/closure-compiler#3973]
TL;DR: a test doesn't appear to verify anything useful. No impact on users.

Hi, I was recently looking at the tests for `LiveVariablesAnalysis` and came across some test cases that appear to be incorrect/useless. 
https://github.com/google/closure-compiler/blob/1b7b32d78640297d8814ddabaf971da530c98659/test/com/google/javascript/jscomp/LiveVariablesAnalysisTest.java#L360-L368
The assertions appear to test that various destructuring parameters aren't escaped by references to the `arguments` array. However, the params in the input code will _never_ be considered escaped because they, and the `arguments` references, are in nested functions; The inputs contain a function e.g. `function f([a]) { arguments; }` but are then wrapped in another function by `computeLiveness`:
https://github.com/google/closure-compiler/blob/1b7b32d78640297d8814ddabaf971da530c98659/test/com/google/javascript/jscomp/LiveVariablesAnalysisTest.java#L677-L678
which then configures `ControlFlowAnalysis` to not visit nested functions (note the lack of `builder.setTraverseFunctions(true)`):
https://github.com/google/closure-compiler/blob/1b7b32d78640297d8814ddabaf971da530c98659/test/com/google/javascript/jscomp/LiveVariablesAnalysisTest.java#L691-L696
This means that the params will never be escaped because:

1. The `arguments` references are never visited during data flow analysis, so [`markAllParametersEscaped`](https://github.com/google/closure-compiler/blob/1b7b32d78640297d8814ddabaf971da530c98659/src/com/google/javascript/jscomp/LiveVariablesAnalysis.java#L433) is never called.
2. Even if `markAllParametersEscaped` were called, the _outer_ params (`param1`, `param2`, `param3`) would be escaped, not the inner params (`a`).

The issue can be shown by adding the following test cases to LiveVariablesAnalysisTest.java:
```java
assertNotEscaped("function f(a) { arguments; }", "a");
   assertEscaped("function f(a) { arguments; }", "a");
```
The param `a` should be escaped because it's a `NAME` param, and therefore `assertNotEscaped` should fail and `assertEscaped` should pass, but the _reverse_ happens.

One way to resolve this would be to add an alternate `assertNotEscaped` helper function that doesn't wrap the input in a function. I believe this would correct the tests so they verify the intended behaviour.

If you agree with the above, I'm happy to open a PR to explore the aforementioned fix.

**Comment by lauraharker** [comment]
Thanks for noticing & reporting this!

I agree these tests are useless as written, and adding a new test helper that doesn't wrap the input seems like the best solution.

(Updating the wrapper `function _FUNCTION(param1, param2 = 1, ...param3)` to have various destructuring params `param4` etc. would work & is more consistent with the other tests in this file, but seems messier)

If you're willing to send a PR, that would be great. I'll assign this issue to you.
